python/desc/sims/GCRCatSimInterface/CatalogClasses.py

```diff
@@ -4,15 +4,91 @@
 from lsst.sims.catalogs.decorators import cached, compound
 from lsst.sims.catUtils.mixins import EBVmixin
 
-__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN"]
+from lsst.sims.catalogs.db import fileDBObject
+
+__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN", "SNFileDBObject",
+           "DC2PhosimCatalogSN"]
 
 #########################################################################
 # define a class to write the PhoSim catalog; defining necessary defaults
 
+class SNFileDBObject(fileDBObject):
+    """
+    Use FileDBObject to provide CatalogDBObject functionality for SN
+    with host galaxies from protoDC2 output to csv files before
+    """
+    dbDefaultValues = {'varsimobjid':-1,
+                       'runid':-1,
+                       'ismultiple':-1,
+                       'run':-1,
+                       'runobjid':-1}
+
+    # These types should be matched to the database.
+    #: Default map is float.  If the column mapping is the same as the
+    # column name, None can be specified
+
+    columns = [('raJ2000', 'snra_in*PI()/180.'),
+               ('decJ2000', 'sndec_in*PI()/180.'),
+               ('Tt0', 't0_in'),
+               ('Tx0', 'x0_in'),
+               ('Tx1', 'x1_in'),
+               ('Tc', 'c_in'),
+               ('id', 'snid_in'),
+               ('Tredshift', 'z_in'),
+               ('redshift', 'z_in'),
+               ('Tgaltileid', 'galtileid')
+              ]
+class DC2PhosimCatalogSN(PhoSimCatalogSN):
+    """
+    Modification of the PhoSimCatalogSN mixin to provide shorter sedFileNames
+    by leaving out the parts of the directory name. Also fix name changes from
+    gamma to shear.
+    """
+    def get_sedFilepath(self):
+        return self.column_by_name('TsedFilepath')
+
+    def get_shorterFileNames(self):
+        """
+        Method to truncate filenames for transient
+        spectra written out by phosim. 
+
+        .. note: the variable sep needs to be in
+        `self.sn_sedfile_prefix` before writing out
+        a phosim catalog.
+        """
+        fnames = self.column_by_name('sedFilepath')
+        sep = 'specFileSN_'
+        split_names = []
+        for fname in fnames:
+            if 'None' not in fname:
+                fname = sep + fname.split(sep)[-1] 
+            else:
+                fname = 'None'
+            split_names.append(fname)
+        return np.array(split_names)
+
+    # column_outputs = PhoSimCatalogSN.column_outputs
+    # column_outputs[PhoSimCatalogSN.column_outputs.index('sedFilepath')] = \
+    #    'shorterFileNames'
+    column_outputs = ['prefix', 'uniqueId', 'raPhoSim', 'decPhoSim',
+                      'phoSimMagNorm', 'shorterFileNames', 'redshift',
+                      'shear1', 'shear2', 'kappa', 'raOffset', 'decOffset',
+                      'spatialmodel', 'internalExtinctionModel',
+                      'galacticExtinctionModel', 'galacticAv', 'galacticRv']
+    cannot_be_null = ['x0', 't0', 'z', 'shorterFileNames']
```

Comment:
Can you put a blank line between `column_outputs` and `cannot_be_null` (for readability)?